src/libsyntax_pos/symbol.rs

```diff
@@ -503,7 +503,7 @@ impl PartialOrd<InternedString> for InternedString {
         if self.symbol == other.symbol {
             return Some(Ordering::Equal);
         }
-        self.with(|self_str| other.with(|other_str| self_str.partial_cmp(&other_str)))
+        self.as_str().partial_cmp(&other.as_str())
```

Comment:
Did you ever get to the bottom of why this is necessary?

Reply:
Apparently it is this bug: https://github.com/rust-lang/rust/issues/48406